extensions/blocks/podcast-player/podcast-player.php

```diff
@@ -0,0 +1,192 @@
+<?php
+/**
+ * Podcast Player Block.
+ *
+ * @since 8.x
+ *
+ * @package Jetpack
+ */
+
+namespace Jetpack\Podcast_Episodes_Block;
+
+use WP_Error;
+use Jetpack_Gutenberg;
+
+const FEATURE_NAME = 'podcast-player';
+const BLOCK_NAME   = 'jetpack/' . FEATURE_NAME;
+const BLOG_SLUG    = 'jetpack-' . FEATURE_NAME;
+/**
+ * Registers the block for use in Gutenberg
+ * This is done via an action so that we can disable
+ * registration if we need to.
+ */
+function register_block() {
+	jetpack_register_block(
+		BLOCK_NAME,
+		array(
+			'attributes'      => array(
+				'url'         => array(
+					'type' => 'url',
+				),
+				'itemsToShow' => array(
+					'type'    => 'integer',
+					'default' => 5,
+				),
+			),
+			'render_callback' => __NAMESPACE__ . '\render_block',
+		)
+	);
+}
+add_action( 'init', __NAMESPACE__ . '\register_block' );
+
+/**
+ * Podcast Player block registration/dependency declaration.
+ *
+ * @param array  $attributes Array containing the Podcast Player block attributes.
+ * @param string $content    String containing the Podcast Player block content.
+ *
+ * @return string
+ */
+function render_block( $attributes, $content ) {
+
+	if ( empty( $attributes['url'] ) ) {
+		return;
+	}
+
+	$track_list = get_track_list( $attributes['url'], $attributes['itemsToShow'] );
+
+	if ( is_wp_error( $track_list ) ) {
+		return '<p>' . __( 'Unable to retrieve track list. Please check your Podcast feed URL.', 'jetpack' ) . '</p>';
+	}
+
+	return render_player( $track_list, $attributes );
+}
+
+/**
+ * Renders the HTML for the Podcast player and tracklist.
+ *
+ * @param array $track_list the list of podcast tracks.
+ * @param array $attributes Array containing the Podcast Player block attributes.
+ * @return string the HTML for the podcast player.
+ */
+function render_player( $track_list, $attributes ) {
+	global $content_width;
+
+	$player_data = array(
+		'type'         => 'audio',
+		// Don't pass strings to JSON, will be truthy in JS.
+		'tracklist'    => true,
+		'tracknumbers' => true,
+		'images'       => true,
+		'artists'      => true,
+		'tracks'       => $track_list,
+	);
+
+	$outer         = 22; // Default padding and border of wrapper.
+	$default_width = 640;
+	$theme_width   = empty( $content_width ) ? $default_width : ( $content_width - $outer );
+
+	// If there are no tracks (it is possible) then display appropriate user facing error message.
+	if ( empty( $track_list ) ) {
+		return '<p>' . __( 'No tracks available to play.', 'jetpack' ) . '</p>';
+	}
+
+	ob_start();
+	wp_playlist_scripts( 'audio' );
+	/**
+	 * Prints and enqueues playlist scripts, styles, and JavaScript templates.
+	 *
+	 * @since 3.9.0
+	 *
+	 * @param string $type  Type of playlist. Possible values are 'audio' or 'video'.
+	 * @param string $style The 'theme' for the playlist. Core provides 'light' and 'dark'.
+	 */
```

Comment:
We don't need this docblock here, it will confuse our Codex parser. Something like this do the trick:
`/** This action is documented in core/src/wp-includes/media.php */`